### DISCUSSION TITLE
Cake Frosting Tasks in Library

### DISCUSSION THREAD

**Discussion by james-a-morgan** [cake-build/cake#3812] [Q&A] [answered]
Hello,

I am trying to create a library of Tasks that I can distribute to my applications as NuGet package.  This way I don't have to duplicate code across my projects.

As soon as I try to move the Tasks into their own library (at this point just another project) they are no longer found when running Cake.

I currently have just a simple Program.cs that calls BuildHost.Run() and references my library.  When I debug, I can see the targets passed in and it walk through the creation of the BuildContext, but once that is done it states it can't find the target.

Is something like this even possible?

Thanks.

James

**Comment by pascalberger** [comment, marked as answer]
This is possible. See [Cake.Frosting.Issues.Recipe](https://github.com/cake-contrib/Cake.Issues.Recipe/tree/develop/Cake.Frosting.Issues.Recipe/Cake.Frosting.Issues.Recipe) for source code of a recipe package which provides some tasks for Frosting builds.

To make tasks of such a library available to your Cake Frosting build you need to register them by adding the following line to the bootstrapping code in the `Main` method of your Cake Frosting project:

```
using System;
using System.Reflection;
using Cake.Frosting;
using MyFrostingLibrary;

public static class Program
{
    public static int Main(string[] args)
    {
        return new CakeHost()
            .UseContext<BuildContext>()
            .AddAssembly(Assembly.GetAssembly(typeof(MyTask)))
            .Run(args);
    }
}
```

**Reply by james-a-morgan** [reply]
Thank you for the fast response.  Once I added that it worked.  I also followed the code here in GitHub and noticed that by default it adds the Entry Assembly on start of the host.

https://github.com/cake-build/cake/blob/a1893a2bba26f49e089d66beeb78421cedf6126d/src/Cake.Frosting/CakeHost.cs#L38

And then the method that you pointed me to adds Assemblies to the under lying list which lets me find my Tasks.

This is going to save me some time.  Thanks again.
